src/main/java/com/mathworks/ci/RunMatlabTestsBuilder.java

```diff
@@ -238,13 +238,19 @@ private synchronized int execMatlabCommand(FilePath workspace, Launcher launcher
         final String uniqueTmpFldrName = getUniqueNameForRunnerFile();
         ProcStarter matlabLauncher;
         try {
+            FilePath genScriptLocation =
+                    getFilePathForUniqueFolder(launcher, uniqueTmpFldrName, workspace);
+
             matlabLauncher = getProcessToRunMatlabCommand(workspace, launcher, listener, envVars,
-                    constructCommandForTest(getInputArguments()), uniqueTmpFldrName);
+                    constructCommandForTest(getInputArguments(), genScriptLocation.getRemote()),
+                    uniqueTmpFldrName);
             
-            // Copy MATLAB scratch file into the workspace.
-            FilePath targetWorkspace = new FilePath(launcher.getChannel(), workspace.getRemote());
+            // Copy MATLAB scratch file into the workspace.      
             copyFileInWorkspace(MatlabBuilderConstants.MATLAB_TESTS_RUNNER_RESOURCE,
-                    MatlabBuilderConstants.MATLAB_TESTS_RUNNER_TARGET_FILE, targetWorkspace);
+                    MatlabBuilderConstants.MATLAB_TESTS_RUNNER_TARGET_FILE, workspace);

```

Comment:
Can we figure out a way not to copy anything to the workspace? Part of the recent work with scriptgen was to eliminate the need to modify the user's workspace. It seems counterproductive to have done all that work and then still put a runner script in the workspace.

The other plugins handle this by running all the code you have in your runner script as the command passed to run_matlab_command. You could also imagine using the function handle trick as well.

Reply:
Done. using UUID approach with `test_runner` as prefix for the file and keeping file in temp folder.